src/assets/Generator.Shared/NextLinkOperationImplementation.cs

```diff
@@ -104,17 +106,49 @@ private void UpdateNextRequestUri(ResponseHeaders headers)
             switch (_headerSource)
             {
                 case HeaderSource.OperationLocation when headers.TryGetValue("Operation-Location", out string? operationLocation):
-                    _nextRequestUri = operationLocation;
+                    _nextRequestUri = AppendOrReplaceApiVersion(operationLocation, _startRequestUri);
                     return;
                 case HeaderSource.AzureAsyncOperation when headers.TryGetValue("Azure-AsyncOperation", out string? azureAsyncOperation):
-                    _nextRequestUri = azureAsyncOperation;
+                    _nextRequestUri = AppendOrReplaceApiVersion(azureAsyncOperation, _startRequestUri);
                     return;
                 case HeaderSource.Location when hasLocation:
-                    _nextRequestUri = location!;
+                    _nextRequestUri = AppendOrReplaceApiVersion(location!, _startRequestUri);
                     return;
             }
         }
 
+        internal static string AppendOrReplaceApiVersion(string uri, Uri startRequestUri)
+        {
+            const string apiVersionQuery = "api-version";
+            NameValueCollection startRequestUriQueries = HttpUtility.ParseQueryString(startRequestUri.Query);
+            var apiVersion = startRequestUriQueries.Get("api-version");
+            if (apiVersion != null)
+            {
+                var uriSpan = uri.AsSpan();
+                if (uri.Contains(apiVersionQuery))
+                {
+                    var apiVersionIndex = uriSpan.IndexOf(apiVersionQuery, StringComparison.OrdinalIgnoreCase);
+                    var indexOfFirstSignAfterApiVersion = uri.IndexOf("&", apiVersionIndex);
+                    ReadOnlySpan<char> uriBeforeApiVersion = uriSpan.Slice(0, apiVersionIndex + apiVersionQuery.Length + "=".Length);
+                    if (indexOfFirstSignAfterApiVersion == -1)
+                    {
+                        return string.Concat(uriBeforeApiVersion, apiVersion);
+                    }
+                    else
+                    {
+                        ReadOnlySpan<char> uriAfterApiVersion = uriSpan.Slice(indexOfFirstSignAfterApiVersion);
+                        return string.Concat(uriBeforeApiVersion, apiVersion, uriAfterApiVersion);
```

Comment:
The test projects use .netstandard 2.0 which doesn't support `string.Concat(ReadOnlySpan<Char>, ReadOnlySpan<Char>, ReadOnlySpan<Char>)`, so this caused build error for test projects.